Libs/DICOM/Widgets/ctkDICOMPatientItemWidget.cpp

```diff
@@ -714,17 +1001,64 @@ void ctkDICOMPatientItemWidget::setSelection(bool selected)
 }
 
 //------------------------------------------------------------------------------
-void ctkDICOMPatientItemWidget::generateStudies()
+void ctkDICOMPatientItemWidget::updateEnabledServersUIFromDB()
 {
   Q_D(ctkDICOMPatientItemWidget);
+  d->updateEnabledServersUIFromDB();
+}
 
-  d->createStudies();
-  if (d->Scheduler && d->Scheduler->getNumberOfQueryRetrieveServers() > 0)
+//------------------------------------------------------------------------------
+bool ctkDICOMPatientItemWidget::askUserActionForServerSecurity()
+{
+  Q_D(ctkDICOMPatientItemWidget);
+  return d->askUserActionForServerSecurity();
+}
+
+//------------------------------------------------------------------------------
+void ctkDICOMPatientItemWidget::generateStudies(bool queryRetrieve)
+{
+  Q_D(ctkDICOMPatientItemWidget);
+
+  if (queryRetrieve)
+  {
+    if (!d->askUserActionForServerSecurity())
+    {
+      return;
+    }
+  }
+
+  d->createStudies(queryRetrieve);
+  if (queryRetrieve && d->Scheduler && d->Scheduler->getNumberOfQueryRetrieveServers() > 0)
```

Comment:
Qt does not use `get` for accessors and uses `count` instead of `numberOf`
```suggestion
  if (queryRetrieve && d->Scheduler && d->Scheduler->queryRetrieveServersCount() > 0)
```